cassandra/metadata.py

```diff
@@ -2529,12 +2542,21 @@ def _query_all(self):
         self.indexes_result = self._handle_results(indexes_success, indexes_result)
         self.views_result = self._handle_results(views_success, views_result)
         # V4-only results
-        self.virtual_keyspaces_result = self._handle_results(virtual_ks_success,
-                                                             virtual_ks_result)
-        self.virtual_tables_result = self._handle_results(virtual_table_success,
-                                                          virtual_table_result)
-        self.virtual_columns_result = self._handle_results(virtual_column_success,
-                                                           virtual_column_result)
+        # These tables don't exist in some DSE versions reporting 4.X so we can
+        # ignore them if we got an error
+        self.virtual_keyspaces_result = self._handle_results(
+            virtual_ks_success, virtual_ks_result,
+            expected_failures=InvalidRequest
+        )
+        self.virtual_tables_result = self._handle_results(
+            virtual_table_success, virtual_table_result,
+            expected_failures=InvalidRequest
+        )
+        self.virtual_columns_result = self._handle_results(
+            virtual_column_success, virtual_column_result,
+            expected_failures=InvalidRequest
+        )
+
```

Comment:
My first idea would have been to simply do a `if virtual_ks_success`... but I think your way makes it more robust. :+1:

Reply:
That was my first thought too, but other places in the codebase check for an `InvalidRequest`, which seems like a good idea, since it maps to "this ks doesn't exist", which is the acceptable case.

Might be worth documenting that in `_handle_results`'s docstring, actually.

Reply:
yep, I like your idea!